scripts/opentitan/autoreg.py

```diff
@@ -91,40 +91,31 @@ def __init__(self, devname: str, ignore_prefix: Optional[str] = None,
         self._log = getLogger('autoreg')
         self._devname = devname
         self._ignore_prefix = ignore_prefix
+        self._git_version = ''
         self._resets = resets or []
         self._parameters: dict[str, Union[int, bool]] = {}
-        self._regwidth = 0
-        self._registers: list[QEMURegister] = []
+        self._regwidth = 0  # bits
+        self._registers: list[AutoRegister] = []
         self._copyright = '@todo Add copyright attributions'
 
     @classproperty
     def generators(cls):
         """Provide a list of supported generators."""
         # pylint: disable=no-self-argument
         actions: list[str] = []
-        for item in cls.__dict__:
-            if not item.startswith('generate_'):
+        prefix = 'generate_'
+        for item in dir(cls):
+            if not item.startswith(prefix):
                 continue
-            actions.append(item[len('generate_'):])
+            actions.append(item.removeprefix(prefix))
         actions.sort()
         return actions
 
-    @classmethod
-    def redent(cls, text: str, spc: int = 0, strip_end: bool = False) -> str:
-        """Utility function to re-indent code string.
-
-           :param text: the text to re-indent
-           :param spc: the number of leading empty space chars to prefix lines
-           :param strip_end: whether to strip trailing whitespace and newline
-        """
-        text = dedent(text.lstrip('\n'))
-        text = indent(text, ' ' * spc)
-        if strip_end:
-            text = text.rstrip(' ').rstrip('\n')
-        return text
-
     def load(self, hfp: TextIO) -> None:
         """Load a register map definition from an OT HJSON file."""
+        self._git_version = retrieve_git_version(hfp.name) or ''
```

Comment:
Nit: if `hfp.name = None`, this will error because `retrieve_git_version` accepts a `str` path and immediately invokes `os.path.abspath` on it. Could also do with some default text here for grammar's sake. Probably needs something instead like:
```suggestion
        self._git_version = "unknown" if hfp.name is None else retrieve_git_version(hfp.name)
```

Reply:
I do not think this condition can occur since hfp is always created by the ArgumentParser, _i.e._ it is known to be a file, not an arbitrary stream, so hfp.name may not be None. This is why I did not bother handling this case, but I guess it should not hurt adding some more code to handle it anyway.